dali/operators/math/normalize/normalize_utils.h

```diff
@@ -0,0 +1,141 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_MATH_NORMALIZE_NORMALIZE_UTILS_H_
+#define DALI_OPERATORS_MATH_NORMALIZE_NORMALIZE_UTILS_H_
+
+#ifdef __SSE__
+#include <xmmintrin.h>
+#endif
+#include <algorithm>
+#include "dali/core/tensor_view.h"
+#include "dali/core/math_util.h"
+#include "dali/pipeline/data/tensor_list.h"
+
+namespace dali {
+namespace normalize {
+
+// Pairwise, elementwise sum of arrays
+template <typename T>
+void SumArrays(T *const *arrays, int num_arrays, int64_t array_size)
+{
+  if (array_size > 1) {
+    int mid = num_arrays >> 1;
+    if (mid > 1)
+      SumArrays(arrays, mid, array_size);
+
+    if (num_arrays - mid > 1)
+      SumArrays(arrays + mid, num_arrays - mid, array_size);
+
+    T *a0 = arrays[0];
+    const T *a1 = arrays[mid];
+    for (int64_t i = 0; i < array_size; i++) {
+      a0[i] += a1[i];
+    }
+  }
+}
+
+/**
+ * @brief Calculates the elementwise sum of all samples in the TensorList and stores it
+ *        in the first sample.
+ *
+ * The function sums the samples pairwise to minimize error. Intermediate results are stored in the
+ * supplied tensor list - only the tlv[0] contains well-defined data after a call to this function.
+ *
+ * @remarks The operation is destructive and the contents of other samples should not be relied
+ *          upon after a call to this function.
+ */
+template <typename T>
+void SumSamples(const TensorListView<StorageCPU, T> &tlv) {
+  int n = tlv.num_samples();
+  if (!n)
+    return;  // empty list - nothing to do
+
+  assert(is_uniform(tlv.shape));
+  auto v = volume(tlv.shape[0]);
+  SumArrays(tlv.data.data(), n, v);
+}
+
+int64_t ReducedVolume(const TensorListShape<> &shape, span<int> axes) {
+  int64_t v = 0;
+  for (int i = 0; i < shape.num_samples(); i++) {
+    auto sample_shape = shape.tensor_shape_span(i);
+    int64_t sample_v = 1;
+    for (auto a : axes)
+      sample_v *= sample_shape[a];
+    v += sample_v;
+  }
+  return v;
+}
+
+
+template <typename T>
+void UniformFill(TensorList<CPUBackend> &tl, const T &value) {
+  T *data = tl.mutable_data<T>();
+  std::fill(data, data + tl.size(), value);
+}
+
+void CalcInvStdDev(const TensorListView<StorageCPU, float> &inv,
+                   const TensorListView<StorageCPU, const float> &stddev,
+                   float scale) {
+  assert(inv.shape == stddev.shape);
+  for (int i = 0; i < inv.shape.num_samples(); i++) {
+    auto v = volume(inv.shape.tensor_shape_span(i));
+    for (int j = 0; j < v; j++) {
+      inv.data[i][j] = stddev.data[i][j] ? scale / stddev.data[i][j] : 0;
+    }
+  }
+}
+
+/**
+ *  @brief Calculates scale/sqrt(x) for nonzero x and keeps 0 when x == 0
+ */
+static void ScaleRSqrtKeepZero(float *data, int64_t n, float scale) {
```

Comment:
The same formula is used in normalize.cc. Can you merge this two pieces of code?